lib/app_profiler/request_parameters.rb

```diff
@@ -16,17 +16,29 @@ def async
       query_param("async")
     end
 
+    def backend
+      query_param("backend") || profile_header_param("backend") ||
+        AppProfiler.backend::NAME
+    end
+
     def valid?
       if mode.blank?
         return false
       end
 
-      unless Parameters::MODES.include?(mode)
-        AppProfiler.logger.info("[Profiler] unsupported profiling mode=#{mode}")
+      return false if backend != AppProfiler::StackprofBackend::NAME && !defined?(AppProfiler::VernierBackend::NAME)
+
+      if defined?(AppProfiler::VernierBackend::NAME) && backend == AppProfiler::VernierBackend::NAME &&
```

Comment:
Maybe we can make this easier by defining `AppProfiler.vernier_supported?`? I think you can use `autoload?` to check if it is loaded or not.